app/filters/LoggingFilter.java

```diff
@@ -0,0 +1,35 @@
+package filters;
+
+import akka.stream.Materializer;
+import java.util.concurrent.CompletionStage;
+import java.util.function.Function;
+import javax.inject.Inject;
+import play.Logger;
+import play.mvc.Filter;
+import play.mvc.Http.RequestHeader;
+import play.mvc.Result;
+
+public class LoggingFilter extends Filter {
+
+    @Inject
+    public LoggingFilter(Materializer mat) {
+        super(mat);
+    }
+
+    @Override
+    public CompletionStage<Result> apply(Function<RequestHeader, CompletionStage<Result>> nextFilter,
+        RequestHeader requestHeader) {
+
+        long startTime = System.currentTimeMillis();
```

Comment:
`System.nanoTime()` is probably better to calculate elapsed time in this case. `System.nanoTime()` is a monotonic clock. If the system changes the time (i.e. NTP time changes), the `System.currentTimeMillis()` method could return negative values. The `System.nanoTime()` method will always return positive values, when used to measure elapsed time.
Also: `System.currentTimeMillis()` can be very imprecise and depends on the operating system. There are operating systems that measure the time in tens of milliseconds. `System.nanoTime()` always uses the most precise clock available to the system.
Use `System.nanoTime()` *only* to measure elapsed time. It is not a clock, it's single values can even be negative, but they will always be monotonic and relatively correct to each other inside a JVM instance.